planner/core/rule_partition_processor.go

```diff
@@ -659,6 +659,10 @@ func (or partitionRangeOR) union(x partitionRangeOR) partitionRangeOR {
 }
 
 func (or partitionRangeOR) simplify() partitionRangeOR {
+	// if the length of the `or` is zero. We should return early.
```

Comment:
If the length of the `or` is zero, is it expected?

Reply:
could you add an example to illustrate in which scenario the length of `or` can be 0?

Reply:
It's expected. For example, we take the test case's conditions `where a > 10 or b is NULL` into consideration. 
1. `a > 10` will not generate the `partitionRange` in `partitionRangeOR`, because of the partition key `a` is less than 5.
2. `b is null` will not generate the `partitionRange` in `partitionRangeOR` too, because the column `b` is not the partition key. 
So the length of `or` is zero. @wjhuang2016 @zz-jason PTAL

Reply:
Ok

Reply:
Actually I mean to add it to the code comment :joy: